test/common/config/grpc_mux_impl_test.cc

```diff
@@ -30,6 +30,8 @@ namespace {
 class GrpcMuxImplTest : public testing::Test {
 public:
   GrpcMuxImplTest() : async_client_(new Grpc::MockAsyncClient()), timer_(new Event::MockTimer()) {
+    // Hack to force linking of the service: https://github.com/google/protobuf/issues/4221
+    envoy::service::discovery::v2::AdsDummy dummy;
```

Comment:
should I include `source/server/configuration_impl.h` here and in `descriptor_test` so that this hack is not sprayed across multiple places? or is there a common include where we can have this hack?

Reply:
Yeah, as Matt says, let's just move this into the main binary, so we no longer need it there. I'm hopeful that we can figure out some `alwayslink` based solution, but for now this is fine. Thanks for taking on this technical debt!

Reply:
Tests are in different binaries with different linkages, so that is expected.